Develop2

Archi1993 opened the pull request.

Archi1993 committed:
add allure report

Archi1993 committed:
Добавлен файлы gitignore и pom

Добавлены классы и тесты для их проверки:

Создание пользователя:
создать уникального пользователя;
создать пользователя, который уже зарегистрирован;
создать пользователя и не заполнить одно из обязательных полей.

Логин пользователя:
логин под существующим пользователем,
логин с неверным логином и паролем.

Изменение данных пользователя:
с авторизацией,
без авторизации,
Для обеих ситуаций нужно проверить, что любое поле можно изменить.

Создание заказа:
с авторизацией,
без авторизации,
с ингредиентами,
без ингредиентов,
с неверным хешем ингредиентов.

Получение заказов конкретного пользователя:
авторизованный пользователь,
неавторизованный пользователь.

Archi1993 committed:
Delete target/allure-results/035b0efa-4512-43a7-83cd-5b21fdc1d7a1-result.json

Archi1993 committed:
Delete target/allure-results/218c6c9e-f7f5-43a2-96c7-5b32ad6c0104-result.json

Archi1993 committed:
Delete target/allure-results/2384f79f-1bbf-44b4-b1fc-3da6fdbe6a5e-result.json

Archi1993 committed:
Delete target/allure-results/2f23b444-27e0-48e8-bc9f-0f255a7ff568-result.json

Archi1993 committed:
Delete target/allure-results/45d5fc09-90de-4420-95fe-06822de9dea5-result.json

Archi1993 committed:
Delete target/allure-results/5f23f187-fd09-4c63-896f-15c68e9e6489-result.json

Archi1993 committed:
Delete target/allure-results/6431016a-7b7b-4deb-a1a3-eafecae0bf82-result.json

Archi1993 committed:
Delete target/allure-results/646ccc83-54d7-484e-b328-48190407d1e8-result.json

Archi1993 committed:
Delete target/allure-results/8aec176c-313d-4b2d-a915-b56bf03d87fd-result.json

Archi1993 committed:
Delete target/allure-results/93baab47-21c8-4635-8941-6b625b8e0252-result.json

Archi1993 committed:
Delete target/allure-results/9eacb883-c98e-4aac-85d3-b109c6b685c3-result.json

Archi1993 committed:
Delete target/allure-results/a62a7e73-7152-4636-b820-40cef1ee312a-result.json

Archi1993 committed:
Delete target/allure-results/aca16f41-bfb8-4ef0-b378-2c128da5077b-result.json

Archi1993 committed:
Delete target/allure-results/aee8bb43-6a5e-43c2-b468-c55ec2ed0772-result.json

Archi1993 committed:
Delete target/allure-results/c0b0f30f-8325-4402-aab0-e69ec8603b96-result.json

Archi1993 committed:
Delete target/allure-results/c16885aa-28b1-4cb7-97a6-456735f670a5-result.json

Archi1993 committed:
Delete target/allure-results/c33614c2-df01-4516-be7f-55f6d0e1546b-result.json

Archi1993 committed:
Delete target/allure-results/da462fd6-9256-4359-a715-1172863f697b-result.json

Archi1993 committed:
Delete target/allure-results/e71c25a2-4b13-4978-af77-389d085623a8-result.json

Archi1993 committed:
Delete target/allure-results/f251f1cd-a515-4df2-9985-2c68978a1904-result.json

pomegranate282922 commented on src/main/java/order/OrderClient.java:
```diff
@@ -0,0 +1,70 @@
+    public ValidatableResponse createOrderAuthorizedUser(String accessToken) {
+        return spec()
+                .header("Authorization", accessToken)
+                .body("{\n\"ingredients\": [\"61c0c5a71d1f82001bdaaa6d\",\"61c0c5a71d1f82001bdaaa70\",\"61c0c5a71d1f82001bdaaa73\"]\n}")
```
Здесь не должно быть тестовых данных. Они должны поступать в метод через параметр.
Нужно исправить везде

pomegranate282922 commented on src/main/java/order/OrderClient.java:
```diff
@@ -0,0 +1,70 @@
+    @Step("Создание заказа неавторизированного пользователя")
+    public ValidatableResponse createOrderUnAuthorizedUser(String accessToken) {
+        return spec()
+                .header("Authorization", accessToken)
```
А токен зачем?
Нужно исправить

pomegranate282922 commented on src/main/java/order/OrderClient.java:
```diff
@@ -0,0 +1,70 @@
+    }
+
+    @Step("Получение заказа авторизованного пользователя")
+    public ValidatableResponse getFamousUserOrders(String accessToken) {
```
Famous?

pomegranate282922 commented on src/main/java/order/OrderClient.java:
```diff
@@ -0,0 +1,70 @@
+    @Step("Получение заказа неавторизованного пользователя")
+    public ValidatableResponse getUnknownUserOrders(String accessToken) {
+        return spec()
+                .header("Authorization", accessToken)
```
Зачем токен?
Нужно исправить

pomegranate282922 commented on src/main/java/specifications/Client.java:
```diff
@@ -0,0 +1,18 @@
+    public class Client {
+        public static final String BASE_URI = "https://stellarburgers.nomoreparties.site/";
+
+        public RequestSpecification spec() {
```
Отлично что используется req spec

pomegranate282922 commented on src/main/java/user/UserGenerator.java:
```diff
@@ -0,0 +1,27 @@
+    protected static String name = "Archi";
+
+
+    public static User random() {
```
Отлично что данные рандомизированы

pomegranate282922 commented on src/test/java/order/CreateOrderTest.java:
```diff
@@ -0,0 +1,104 @@
+        var creds = Credentials.from(user);
+        ValidatableResponse loginResponse = clientUser.loginUser(creds);
+        accessToken = checkUser.loggedInSuccessfully(loginResponse);
+        assertThat("Failed to login!", accessToken, is(notNullValue()));
```
Слишком много проверок, не относящихся к тесту. Здесь ты создаешь заказ. Доп проверки про юзера сейчас нас не интересуют - для них будут же свои тесты.
Нужно исправить везде

pomegranate282922 commented on src/test/java/order/CreateOrderTest.java:
```diff
@@ -0,0 +1,104 @@
+
+        ValidatableResponse orderResponse = clientOrder.createOrderAuthorizedUser(accessToken);
+        number = checkOrder.checkCreatedOrderSuccessfully(orderResponse);
+        assert number != 0;
```
Используем ассерт из junit 
Нужно исправить

pomegranate282922 commented on src/test/java/order/CreateOrderTest.java:
```diff
@@ -0,0 +1,104 @@
+    @DisplayName("Создание заказа неавторизованного пользователя")
+    @Description("Проверка полученного ответа при создании заказа неавторизованного пользователя")
+    public void createNewUnSuccessfullyOrder() {
+        var user = UserGenerator.random();
```
Зачем создавать юзера для теста неавторизованного пользователя? 
нужно исправить везде

pomegranate282922 commented on src/test/java/order/CreateOrderTest.java:
```diff
@@ -0,0 +1,104 @@
+    @After
+    public void deleteUser() {
+        ValidatableResponse delete = clientUser.deleteUser(accessToken);
+        checkUser.deletedSuccessfully(delete);
```
Тоже не нужна проверка 
Нужно исправить

pomegranate282922 commented on src/test/java/order/GetOrderTest.java:
```diff
@@ -0,0 +1,86 @@
+        ValidatableResponse delete = clientUser.deleteUser(accessToken);
+        checkUser.deletedSuccessfully(delete);
+    }
+
```
Здесь применить все замечания из предыдущего класса 
Нужно исправить

pomegranate282922 commented on src/test/java/order/GetOrderTest.java:
```diff
@@ -0,0 +1,86 @@
+
+    @Before
+    public void setFlagAfter() {
+        flag = true;
```
Зачем это?
Нужно исправить

pomegranate282922 commented on src/test/java/user/CreateUserTest.java:
```diff
@@ -0,0 +1,104 @@
+
+    @After
+    public void deleteUser() {
+        if (flag) {
```
Вместо флага можно проверять на непустоту токен 
Можно улучшить

pomegranate282922 reviewed (commented):
Работа хорошая! Все тесты реализованы. Есть рандомизация. Все юзеров удаляются в конце тестов. 
замечания:
1. Нужно убрать все проверки, не относящиеся к данному тесту 
2. Нужно убрать все создания юзеров, когда они не нужны в тесте 
3. Использовать ассерт из junit 
4. Убрать тестовые данные из вызовов ручек 
5. Нужен аллюр отчет 
6. Остальное по мелочи

Archi1993 committed:
1) В OrderClient удалил все тестовые данные, сделал реализацию через параметр
2) Удалил использоваение токенов, где они не нужны
3) В OrderClient переименовал getFamousUserOrders на getAuthorizedUserOrders
4) В GetOrderTest оставил flag чтобы уменьшить количество проверок в тесте
5) Удалил создания юзеров, когда они не нужны в тесте
6) Использовать ассерт из junit
7) Добавил аллюр отчет

Archi1993 committed:
Merge remote-tracking branch 'origin/develop2' into develop2

Archi1993 committed:
add allure report

pomegranate282922 commented on src/main/java/order/OrderClient.java:
```diff
@@ -0,0 +1,70 @@
+public class OrderClient extends Client {
+    private static final String CREATE_ORDER_ENDPOINT = "/api/orders";
+    private static final String GET_ORDER_ENDPOINT = "/api/orders";
+    private static final String allCorrectIngredient = "{\n\"ingredients\": [\"61c0c5a71d1f82001bdaaa6d\",\"61c0c5a71d1f82001bdaaa70\",\"61c0c5a71d1f82001bdaaa73\"]\n}";
```
Static final поля - классовые константы - именуем заглавными буквами через нижнее подчеркивание

pomegranate282922 commented on src/main/java/order/OrderClient.java:
```diff
@@ -0,0 +1,70 @@
+    @Step("Создание заказа неавторизированного пользователя")
+    public ValidatableResponse createOrderUnAuthorizedUser() {
+        return spec()
+                .body(allCorrectIngredient)
```
Данные надо передавать через параметр, а не хардкодить.
Что если тебе надо протестировать другие данные? Новый метод делать? Это нехорошо.
Нужно исправить

pomegranate282922 commented on src/main/java/order/OrderClient.java:
```diff
@@ -0,0 +1,70 @@
+    }
+
+    @Step("Создание заказа c неверным хешем ингредиентов")
+    public ValidatableResponse createOrderWithUnknownIngredients(String accessToken) {
```
По сути этот метод никак не отличается от предыдущего. Только данными.
Можно улучшить

pomegranate282922 commented on src/test/java/user/ChangeUserSettingsTest.java:
```diff
@@ -0,0 +1,165 @@
+        ValidatableResponse delete = clientUser.deleteUser(accessToken);
+        checkUser.deletedSuccessfully(delete);
+    }
+}
```
Не исправлено замечание о многих ненужных проверках в тестах. Нужно исправить.

pomegranate282922 replied:
Написала для примера в одном тесте - нужна проверка того, для чего собственно написан тест. А не для каждого его шага. 
нужно исправить

pomegranate282922 commented on src/test/java/user/ChangeUserSettingsTest.java:
```diff
@@ -0,0 +1,165 @@
+    @After
+    public void deleteUser() {
+        ValidatableResponse delete = clientUser.deleteUser(accessToken);
+        checkUser.deletedSuccessfully(delete);
```
Эта проверка не нужна

pomegranate282922 commented on src/test/java/user/ChangeUserSettingsTest.java:
```diff
@@ -0,0 +1,165 @@
+    public void notSuccessfullyUpdateNameUser() {
+        var user = UserGenerator.random();
+        ValidatableResponse response = clientUser.createUser(user);
+        checkUser.createdSuccessfully(response);
```
Эта проверка не нужна

pomegranate282922 commented on src/test/java/user/ChangeUserSettingsTest.java:
```diff
@@ -0,0 +1,165 @@
+        var creds = Credentials.from(user); //Требуется корректная авторизация пользователя, для его успешного удаления после теста
+        ValidatableResponse loginResponse = clientUser.loginUser(creds);
+        accessToken = checkUser.loggedInSuccessfully(loginResponse);
+        assertThat("Failed to login!", accessToken, is(notNullValue()));
```
Эта проверка не нужна

pomegranate282922 reviewed (commented):
Многое исправил, но пара замечаний осталась. Нужно доделать.

Archi1993 committed:
1) Исправил константы
2) Данные передаю через параметры
3) Удалил ненужные проверки из тестов user

Archi1993 committed:
1) Добавил новый класс OrderData
2) Изменил классы Order и OrderCleint
3) Изменил тестовые классы order

pomegranate282922 reviewed (approved):
Вот! Другое дело! Зачет!
